processor/src/main/java/org/derive4j/processor/derivator/LazyConstructorDerivator.java

```diff
@@ -73,20 +73,20 @@
                 .addModifiers(Modifier.PRIVATE)
                 .returns(typeName)
                 .addCode(CodeBlock.builder()
-                    .addStatement("$T _evaluation = this.evaluation", typeName)
-                    .beginControlFlow("if (_evaluation == null)")
-                    .beginControlFlow("synchronized (this.lock)")
```

Comment:
Suppliers.memoize also implements Serializable (without serializing the cached value) which limit the possible implementation much more than if we don't implement Serializable. I think using an initialized field is therefore not necessary if we don't implement Serializable. Benchmarking synchronization on ```this``` is worth it, though.